src/main/java/nl/tudelft/contextproject/gui/CreateScriptController.java

```diff
@@ -1,120 +1,227 @@
 import javafx.fxml.FXML;
 import javafx.fxml.FXMLLoader;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Alert.AlertType;
 import javafx.scene.control.Button;
+import javafx.scene.control.ButtonType;
 import javafx.scene.control.ChoiceBox;
+import javafx.scene.control.TableCell;
 import javafx.scene.control.TableColumn;
 import javafx.scene.control.TableColumn.CellDataFeatures;
+import javafx.scene.control.TableColumn.CellEditEvent;
 import javafx.scene.control.TableView;
 import javafx.scene.control.TextField;
 import javafx.scene.control.cell.PropertyValueFactory;
+import javafx.scene.control.cell.TextFieldTableCell;
 import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.HBox;
 import javafx.util.Callback;
 
 import main.java.nl.tudelft.contextproject.ContextTFP;
 import main.java.nl.tudelft.contextproject.camera.Camera;
 import main.java.nl.tudelft.contextproject.camera.CameraSettings;
 import main.java.nl.tudelft.contextproject.presets.InstantPreset;
-import main.java.nl.tudelft.contextproject.script.Script;
 import main.java.nl.tudelft.contextproject.script.Shot;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 /**
  * Controller class for the script creation screen.
+ * This class is responsible for the functions of the create script screen, to
+ * allow for easy creation and saving of a new script.
  * 
- * @author Steven Meijer
+ * @since 0.2
  */
 public class CreateScriptController {
 
-    Script script;
-
     @FXML private Button btnAdd;
     @FXML private Button btnBack;
 
     @FXML private ChoiceBox<Integer> addCamera;
     @FXML private ChoiceBox<Integer> addPreset;
 
+    @FXML private HBox paneEdit;
+
     @FXML private TableView<Shot> tableEvents;
-    @FXML private TableColumn<Shot, String> tAdd;
-    @FXML private TableColumn<Shot, Integer> tCamera;
-    @FXML private TableColumn<Shot, String> tDescription;
-    @FXML private TableColumn<Shot, Integer> tID;
-    @FXML private TableColumn<Shot, Integer> tPreset;
-    @FXML private TableColumn<Shot, String> tShot;
+    @FXML private TableColumn<Shot, Shot> columnAction;
+    @FXML private TableColumn<Shot, Integer> columnCamera;
+    @FXML private TableColumn<Shot, String> columnDescription;
+    @FXML private TableColumn<Shot, Integer> columnID;
+    @FXML private TableColumn<Shot, Integer> columnPreset;
+    @FXML private TableColumn<Shot, String> columnShot;
 
     @FXML private TextField addShot;
     @FXML private TextField addDescription;
 
-    @FXML
-    private void initialize() {
+    /**
+     * Initialize method used by JavaFX.
+     */
+    @FXML private void initialize() {
 
         //TEMP
         Camera c = new Camera();
+        new Camera();
         c.addPreset(new InstantPreset(new CameraSettings(), 0));
         c.addPreset(new InstantPreset(new CameraSettings(), 1));
         c.addPreset(new InstantPreset(new CameraSettings(), 2));
         //
 
```

Comment:
After the merge we should move this to ContextTFP.java in order to prevent the creation of duplicate cameras.